cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/hybrid_transfers.rs

```diff
@@ -819,3 +819,88 @@ fn transfer_native_asset_from_relay_to_para_through_asset_hub() {
 	// should be non-zero
 	assert!(receiver_assets_after < receiver_assets_before + amount_to_send);
 }
+
+// ==============================================================================================
+// ==== Bidirectional Transfer - Native + Teleportable Foreign Assets - Parachain<->AssetHub ====
+// ==============================================================================================
+/// Transfers of native asset plus teleportable foreign asset from Parachain to AssetHub and back
+/// with fees paid using native asset.
+#[test]
+fn bidirectional_transfer_multiple_assets_between_penpal_and_asset_hub() {
+	fn execute_xcm_penpal_to_asset_hub(t: ParaToSystemParaTest) -> DispatchResult {
+		let all_assets = t.args.assets.clone().into_inner();
+		let mut assets = all_assets.clone();
+		let mut fees = assets.remove(t.args.fee_asset_item as usize);
+		// TODO(https://github.com/paritytech/polkadot-sdk/issues/6197): dry-run to get exact fees.
+		// For now just use half the fees locally, half on dest
+		if let Fungible(fees_amount) = fees.fun {
+			fees.fun = Fungible(fees_amount / 2);
+		}
+		// xcm to be executed at dest
+		let xcm_on_dest = Xcm(vec![
+			// since this is the last hop, we don't need to further use any assets previously
+			// reserved for fees (there are no further hops to cover transport fees for); we
+			// RefundSurplus to get back any unspent fees
+			RefundSurplus,
+			DepositAsset { assets: Wild(All), beneficiary: t.args.beneficiary },
+		]);
+		let xcm = Xcm::<()>(vec![
+			WithdrawAsset(all_assets.into()),
+			PayFees { asset: fees.clone() },
+			InitiateTransfer {
+				destination: t.args.dest,
+				remote_fees: Some(AssetTransferFilter::ReserveWithdraw(fees.into())),
```

Comment:
> Furthermore, through remote_fees: Option<AssetTransferFilter>, it shall allow specifying a single asset to be used for fees on dest chain. 

Can `remote_fees` contains multiple assets and also to allow `PayFees` with multiple assets?

Reply:
No, this was discussed in the ‘PayFees’ RFC and decided to keep it simple with single asset. There aren’t real examples of paying multiple fees in different assets on a single chain.

Reply:
> There aren’t real examples of paying multiple fees in different assets on a single chain.

Fee 1 in DOT dry run [here](https://github.com/yrong/polkadot-sdk/blob/0e27a22dd6a7af3130393841b0d67ca71fbf0035/bridges/snowbridge/pallets/outbound-queue-v2/runtime-api/src/lib.rs#L21) for the execution cost(i.e. merkle the message tree) on BH.

Fee 2 in WETH to dry run [submit](https://github.com/Snowfork/snowbridge/blob/fa9b8a7b3551cc9df8ee3239dc2830be5d84728e/contracts/src/Gateway.sol#L419) on Ethereum. 

We need these two kinds of fees as user input for building the transfer [xcm](https://github.com/yrong/polkadot-sdk/blob/0e27a22dd6a7af3130393841b0d67ca71fbf0035/cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_v2.rs#L271) on AH.

Reply:
But the XCM "executed" on Ethereum and the one executed on BH are different and can have different fee assets:

Let's assume a simple WETH transfer from AH to eth (the XCM intended for Ethereum is actually irrelevant):
```rust
XcmAtEthereum = Xcm::new(vec![
	WithdrawAsset(WETH, 10),
	PayFees(WETH, 0.01),
	DepositAsset(Wild(All), beneficiary),
]);
```

The XCM executing on BH is actually (assuming current `SovereignPaidExporter`):
```rust
XcmAtBridgeHub = Xcm::new(vec![
	WithdrawAsset(DOT, 0.01),
	PayFees(DOT, 0.01),
	ExportMessage(XcmAtEthereum),
	RefundSurplus,
	DepositAsset(Wild(All), origin),
]);
```

So you can see that BH is using `PayFees(DOT)` and Ethereum is using `PayFees(WETH)`.

----

AH executor is not aware of BH, so AH "sends directly" to Ethereum.

On AH, the instructions to be executed would have to be:

```rust
XcmAtAssetHub = Xcm::new(vec![
	WhateverInstructionsBefore,
	...,
	InitiateTransfer(
		destination: BridgeHub,
		assets: vec![AssetTransferFilter::DestinationReserve(WETH, 9.99)],
		remote_fees: Some(AssetTransferFilter::DestinationReserve(WETH, 0.01)),
		preserve_origin: false,
		remote_xcm: Xcm::new(vec![DepositAsset(Wild(All), beneficiary)]),
	),
]);
```

Then the AH exporter knows it needs to send to BH, builds `XcmAtBridgeHub` from above, charges `0.01` DOT transport fee and sends to BH.

Reply:
But it's never two assets for fees in one chain, it's one asset in two different chains

Reply:
> charges 0.01 DOT transport fee and sends to BH with `SovereignPaidExporter`

The 0.01 DOT is only to cover the execution of xcm on BH until delivered to message queue [here](https://github.com/yrong/polkadot-sdk/blob/0e27a22dd6a7af3130393841b0d67ca71fbf0035/bridges/snowbridge/pallets/outbound-queue-v2/src/send_message_impl.rs#L53)

After that the message is async dispatched on idle with [committed to header](https://github.com/yrong/polkadot-sdk/blob/0e27a22dd6a7af3130393841b0d67ca71fbf0035/bridges/snowbridge/pallets/outbound-queue-v2/src/lib.rs#L344). This should be the main cost of the execution on BH and it's what the dry run [here](https://github.com/yrong/polkadot-sdk/blob/0e27a22dd6a7af3130393841b0d67ca71fbf0035/bridges/snowbridge/pallets/outbound-queue-v2/runtime-api/src/lib.rs#L21) estimated for.

Say it's 0.1 DOT(10x more), so the intention here is not to mix the two portions of fees together(i.e. 0.1+0.01=0.11 DOT), we minimize the static config of delivery fee [here](https://github.com/yrong/polkadot-sdk/blob/0e27a22dd6a7af3130393841b0d67ca71fbf0035/cumulus/parachains/runtimes/assets/asset-hub-westend/src/xcm_config.rs#L694) as 0.01 DOT and make the 0.1 DOT a user input.

Anyway, we can workaround it with the single fee asset. UI just convert the 0.1 DOT to WETH and added that to the [remote fee portion](https://github.com/yrong/polkadot-sdk/blob/0e27a22dd6a7af3130393841b0d67ca71fbf0035/cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_v2.rs#L271) when building xcm on AH.